### ISSUE TITLE
Allow creating shared wineprefixes

### ISSUE THREAD

**Issue by DragoonAethis** [ValveSoftware/Proton#604]
Currently, Proton creates a separate wineprefix for each and every game - while this is very good for most titles, there are some that can read saves from other titles and change their behavior based on that. For example, you can carry over your character and choices in Mass Effect or The Witcher.

**Comment by nstgc** [comment]
You can symlink your user data. For Mass Effect, for instance, I have it pointing at the bioware directory in my home directory (~/Documents/My\ Games/Bioware if memory serves me).

**Comment by DragoonAethis** [comment]
That's certainly possible, but many future Steam Play users (once it goes out of beta) won't be able to do this (either because they lack skills to do so, knowledge of which files are needed to be symlinked where, or lack of out-of-box terminal access on devices like Steam Machines). This should be possible to specify per-title to make Steam Play a bit more transparent to the end user.

**Comment by nstgc** [comment]
True enough. . .I think the real solution is to automatically symlink steamuser to ~/. . .or something like that.

**Comment by thunder1410** [comment]
Maybe make a sysmlink the (my) documents folder for every game to for example /home/.steam/steamplay/MyDocuments?

**Comment by hmlendea** [comment]
Just dropped here to show my support for having this available out-of-the-box.

TL;DR Symlinks and manually copying files is not always an option, or as easy as it might sound. Others already mentioned this, but here I come with another example.

A few days ago I wanted to play The Walking Dead: A New Frontier (Season 3) on the TV with Remote Play (Steam Link) while being in my hometown for a few weeks and had to go through the pain of:
 - Opening the laptop I brought with me just to copy the save
 - Connecting to my home VPN so that I can access the local (home) network
 - Copying the files via SSH to my desktop back home

And it all worked out because of luck. Thank God I had SSH configured for my local network back home, and a VPN to access that network.
The average user wouldn't have known how to do this, and I'm pretty sure only a handful of people have all this infrastructure in place already.

Oh, and I would have had to do it for Season 2 as well, but thankfully my old laptop is "powerful" enough to handle a game that old, so even tho it spewed fire I still preffered it over copying the files remotely. But I'm gonna have to do it for Season 4 soon.

Now if the prefix for all 4 TWD games was shared, all I would have had to do would have been to simply install S2 alongside S3, et voilà. All doable quickly and easily within the Steam Link app and using just the gamepad.

**Comment by jebez1** [comment]
The solution:
LAUNCH OPTIONS: STEAM_COMPAT_DATA_PATH=proton_prefix_path %command%

See the link above.

**Comment by SomeoneIsWorking** [comment]
> True enough. . .I think the real solution is to automatically symlink steamuser to ~/. . .or something like that.

One of the things I don't understand about separate prefixes is, in a Windows PC, you install all the games in the same system so why is the default behavior creating separate systems?

I know about that some prefixes are 32bit and some are 64bit and some prefixes require some library configurations but honestly these issues should be resolved instead of keeping the current architecture...

**Comment by willrabbermann** [comment]
I have been running only two prefixes for my games in wine outside of steam-for-linux. Works flawlessly. I don't understand why we need to reinstall c++ and directx for every game when directx 11 is backwards compatbile and you can install 2005-2022 c++ no problem. The only issue is with .NET 4.8 and .NET 3.5 SP1 as one will overwrite the other's Common Language Runtime (CLR) mscoree.dll. So it looks like you only need two prefixes minimum, one with 3.5 SP1 and a modern one with 4.8. Or at the very least symlink immutable DLL's.

**Comment by ziipzaaapM16A4** [comment]
I agree.
I am new to Nobara and Proton and i would like it very much if i can have a single storage for all prefix paths.

How about a Default Prefix path and if a game isn't working correctly the user can "sandbox" it by adding via function/button click a new prefix for a specific game. this honsestly would make so many things easier as the user wants to search for his files at one location. not at 50 and also having to find the app id first.


EDIT:
I would like to refine my previous answer after giving it some more thought. I believe it would be much more practical if we had a shared prefix specifically for the Documents and AppData folders.

To my knowledge, there are no programs that install core dependencies into these specific directories. If something goes wrong, it would be relatively easy to remove a specific game’s folder from this shared space. This approach would bridge the gap between dependency handling (system stability) and having a central, known location for save games and mods.

Of course, there is still the possibility that files and dependencies from launchers will be stored in AppData, but this is rarely the case for the Documents folder.

**Comment by VladimirMrzv** [comment]
a lot of games downloading the same file again

<img width="1945" height="945" alt="Image" src="https://github.com/user-attachments/assets/3b785ea0-7441-49b8-a295-74f54cb6a783" />

<img width="1007" height="153" alt="Image" src="https://github.com/user-attachments/assets/292eee49-5a51-4ded-8c57-cc651fd2cac0" />

**Comment by NTMan** [comment]
I think #9783 isn't really #604 — they're different layers, and the relevant one here is something #604 can't reach.

#604 shares a wineprefix (compatdata). #9783 is about the Steam client downloading the same launcher installer depot once per appid — that's the content layer, before Wine is ever involved.

Checkable on SteamDB: the Ubisoft Connect installer is a single depot ([1716751](https://steamdb.info/app/1716750/linked/), ~252 MB) referenced by 100+ apps via depotfromapp, but it is not on the shared-install path that Steamworks Common Redistributables (228980) uses — so it's fetched and patched per appid. A shared prefix would dedup the installed launcher, not that download.

So, as a question rather than a complaint: wasn't "download once and link to all referencing apps" — the 228980 model the OP referenced — the actual point of #9783? That's a depot/content behavior, and #604 can't deliver it regardless of how prefix-sharing lands. Is the depot-level dedup tracked anywhere separately?

**Comment by kisak-valve** [comment]
Ubisoft's configuration of their shared depot would be something for them to evaluate outside of Proton. That's OS independent.

It looks like my initial assessment of #9783 was from the perspective the parallel installed Ubisoft Connect instances and hypothetically having that linked together for update cycles between games.
